### PR TITLE
fix(start): adding support for streaming using `renderToReadableStream`

### PR DESCRIPTION
**Pull request by lithdew**
react-dom/server does not export renderToPipeableStream for JavaScript runtimes such as Bun.

### CODE REVIEW

**Comment by lithdew** [comment]
CI appears to be failing due to some package exports that aren't touched in this PR + some unconfigured CI secrets.

File: packages/start/src/server/defaultStreamHandler.tsx [hunk 1]
```diff
@@ -16,30 +16,54 @@ export const defaultStreamHandler: StartHandler<AnyRouter> = async ({
...
-  })
-
-  return new Response(passthrough as any, {
-    status: router.state.statusCode,
-    statusText: `${router.state.statusCode}`.startsWith('5')
-      ? 'Internal Server Error'
-      : 'OK',
-    headers: responseHeaders,
-  })
+  if (typeof ReactDOMServer.renderToReadableStream === 'function') {
```
Comment (schiller-manuel) [inline packages/start/src/server/defaultStreamHandler.tsx:19]: `renderToReadableStream` now has precedence over `renderToPipeableStream`, do we want this?

Comment (lithdew) [reply packages/start/src/server/defaultStreamHandler.tsx:19]: For environments where `renderToReadableStream` is available (environment supporting Web Streams API i.e. the browser, Bun, etc.), `renderToPipeableStream` is not available.

See:

https://github.com/facebook/react/blob/main/packages/react-dom/package.json#L60-L70
https://github.com/facebook/react/blob/main/packages/react-dom/server.browser.js
https://github.com/facebook/react/blob/main/packages/react-dom/server.node.js
https://github.com/facebook/react/blob/main/packages/react-dom/server.edge.js
https://github.com/facebook/react/blob/main/packages/react-dom/server.bun.js

Comment (schiller-manuel) [reply packages/start/src/server/defaultStreamHandler.tsx:19]: ah, it's either - or. I assumed that an implementation could have both.

**Pull request merged**